arithmetic_analysis/secant_method.py

```diff
@@ -1,28 +1,48 @@
 # Implementing Secant method in Python
+# https://en.wikipedia.org/wiki/Secant_method
 # Author: dimgrichr
 
 
 from math import exp
+from typing import Callable
 
+RealFunc = Callable[[float], float]  # type alias for real -> real function
 
-def f(x):
+
+def secant_method(
+    func: RealFunc, lower_bound: float, upper_bound: float, repeats: int
+) -> float:
+    """
+    >>> secant_method(lambda x: 8 * x - 2 * exp(-x), 1, 3, 2)
+    0.2139409276214589
+    >>> secant_method(lambda x: x ** 3 - 2 * x - 5, -3, 3, 10)
+    2.094546116969569
+    >>> secant_method(lambda x: x ** 3 - 1, 0, 5, 5)
+    0.07992494940223759
+    >>> secant_method(lambda x: x ** 3 - 1, 0, 5, 500)
+    Traceback (most recent call last):
+    ...
+    ZeroDivisionError: Could not find root
+    """
+    lower: float = lower_bound
+    upper: float = upper_bound
+    for _ in range(repeats):
+        value: float = func(upper)
+        numerator: float = value * (upper - lower)
+        denominator: float = value - func(lower)

```

Comment:
```suggestion
        value = func(upper)
        numerator = value * (upper - lower)
        denominator = value - func(lower)
```
Type hints on local variables are not needed and they create a lot of visual and cognitive clutter.  Python is smart enough to figure out that func() returns a float so value is a float, etc.

Reply:
Made the changes